### DISCUSSION TITLE
Does one need to enable I2C or I2S on Raspberry before they will work?

### DISCUSSION THREAD

**Discussion by iluvsa** [probonopd/MiniDexed#700] [Q&A] [answered]
This might be a silly one, but I have a Dexxed running perfectly using a Raspberry Pi 4 board and the audio on board, but when I tried to add hardware (oled, dac) and changed the minidexed.ini file accordingly, neither seem to work - i know the .ini changes are working because I now get no audio out of the original jack on board. Am I supposed to enable I2C and I2S protocols somewhere else, or should the changes to the .ini file enable those automatically?! Thanks!

**Comment by blixabloxa** [comment]
Did you also enable the LCDI2CAddress? This is only required if you have the display with the piggyback board.

LCDEnabled=1
LCDI2CAddress=0x27

Also, another issue I had with my display was that it came from the factory with an incorrect brightness setting, making the display look like it wasn't working. I needed to adjust the POT to get it to display properly.

**Reply by iluvsa** [reply]
Hey! This was an OLED 128x32 display, but I did enable as above, except the address was 0x3c for the I2C. This is a brand new Pi out of the box so I am wondering if I have to enable the i2c in the bios or command line?! It’s too weird that the DAC and display don’t work!

**Reply by blixabloxa** [reply]
As far as I know, the only config for the display is in the ini file. Maybe someone with an OLED screen can chime in here.

**Reply by probonopd** [reply]
Only `minidexed.ini` needs to be edited, nothing else.
Can you get the display to work with software other than MiniDexed?

**Reply by iluvsa** [reply]
Hey!

I have’t tried any other software and it’s a brand new Pi - waiting on a micro-HDMI so I can see what’s happening!! (no screen at the moment) - so you are saying that I shouldn’t have to enable I2C interface via command line before attempting to get the OLED to work? Thanks for your help!!

**Reply by probonopd** [reply]
Well, you need to configure `minidexed.ini`. There is no command line in MiniDexed.

**Reply by iluvsa** [reply]
Ok I have confirmed the display is getting 5V and each of the SDA and SCK pins are reading 3.3, so I must have screwed something up in the .ini!! Multiple displays aren’t working so I don’t think it’s a faulty unit!

**Reply by iluvsa** [reply, marked as answer]
Ok nevermind got it!! I didn’t realize you have to leave the “LCDEnabled=1”command in there even when using a OLED! Everything’s working great now!!!
